flask_monitoringdashboard/core/exceptions/exception_logger.py

```diff
@@ -0,0 +1,109 @@
+from types import TracebackType
+from typing import Union
+
+from sqlalchemy.orm import Session
+from flask_monitoringdashboard.core.exceptions.scoped_exception_logger import (
+    ScopedExceptionLogger,
+)
+from flask_monitoringdashboard.core.exceptions.stack_frame_parsing import (
+    get_function_definition_from_frame,
+)
+from flask_monitoringdashboard.core.exceptions.stack_trace_hashing import (
+    hash_stack_trace,
+)
+from flask_monitoringdashboard.database.exception_occurrence import add_exception_occurrence
+from flask_monitoringdashboard.database.stack_trace_snapshot import (
+    add_stack_trace_snapshot,
+    get_stack_trace_by_hash,
+)
+from flask_monitoringdashboard.database.exception_stack_line import (
+    add_exception_stack_line,
+)
+from flask_monitoringdashboard.database.exception_frame import add_exception_frame
+from flask_monitoringdashboard.database.function_location import add_function_location
+from flask_monitoringdashboard.database.file_path import add_file_path
+from flask_monitoringdashboard.database.function_definition import (
+    add_function_definition,
+)
+from flask_monitoringdashboard.database.exception_message import add_exception_message
+from flask_monitoringdashboard.database.exception_type import add_exception_type
+
+
+class ExceptionLogger:
+
+    def __init__(self, scoped_logger: ScopedExceptionLogger):
+        self.user_captured_exceptions: list[BaseException] = (
+            scoped_logger.user_captured_exceptions
+        )
+        self.uncaught_exception: Union[BaseException, None] = (
+            scoped_logger.uncaught_exception
+        )
+
+    def _save_to_db(
```

Comment:
this method could be static, or could even be a function - it does not refer to the `self` variable at all.